docker/build.sh

```diff
@@ -1,17 +1,28 @@
 #!/bin/bash
 set -e
 
+sudo apt-get update
+sudo apt-get install -y flatpak-builder git bzip2 git \
+    python3 python3-requirement-parser python3-toml
+
 git config --global protocol.file.allow always
 
 flatpak remote-add --user --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo
 [[ ! -d org.flatpak.Builder ]] && git clone https://github.com/flathub/org.flatpak.Builder
 
-sudo apt-get install -y python3-requirement-parser python3-toml
 mv flatpak-builder-lint-deps.json org.flatpak.Builder/flatpak-builder-lint-deps.json
 python3 rewrite-manifest.py
 
+case $1 in
+    amd64)
+        arch=x86_64
+        ;;
+    arm64)
+        arch=aarch64
+        ;;
+esac
+
 cd org.flatpak.Builder 
-flatpak-builder --user --verbose --force-clean --repo=repo \
-    --ccache --install-deps-from=flathub \
-    builddir org.flatpak.Builder.json
-rm -rf files/lib/debug
+flatpak-builder --arch=$arch --user --verbose --force-clean --repo=repo \
```

Comment:
How does it work? Does the manifest have a cross-compilation toolchain set up correctly? Or does it use QEMU as well to execute binaries from the SDK in the target architecture?

Reply:
There is no cross-compilation with flaptak (ie compiler running on a platform that is not the target). It will run everything in QEMU.